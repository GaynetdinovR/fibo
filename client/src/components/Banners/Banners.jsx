import styles from "../../styles/components/Banners.module.sass";
import "react-multi-carousel/lib/styles.css";
import Carousel from "react-multi-carousel";

const Banners = () => {
	const banners = [
		{ src: "./content/banner_1.jpg" },
		{ src: "./content/banner_3.jpg" },
		{ src: "./content/banner_2.jpg" }
	];

	const responsive = {
		desktop: {
			breakpoint: { max: 3000, min: 1024 },
			items: 3,
			slidesToSlide: 1
		},
		tablet: {
			breakpoint: { max: 1024, min: 464 },
			items: 2,
			slidesToSlide: 1
		},
		mobile: {
			breakpoint: { max: 768, min: 0 },
			items: 1,
			slidesToSlide: 1
		}
	};

	return (
		<Carousel
			swipeable={true}
			draggable={false}
			showDots={false}
			responsive={responsive}
			ssr={true}
			infinite={true}
			autoPlay={true}
			autoPlaySpeed={4000}
			keyBoardControl={true}
			containerClass={styles.banners__carousel}
		>
			{banners.map((banner, i) => (
				<div key={i} className={styles.banners__banner}>
					<img src={banner.src} alt="banner" />
				</div>
			))}
		</Carousel>
	);
};

export default Banners;