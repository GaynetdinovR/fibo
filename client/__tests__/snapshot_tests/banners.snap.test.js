import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import Banners from "../../src/components/Banners/Banners.jsx";

it("renders correctly", () => {
	const tree = renderer.create(<Banners/>).toJSON();
	expect(tree).toMatchSnapshot();
});
