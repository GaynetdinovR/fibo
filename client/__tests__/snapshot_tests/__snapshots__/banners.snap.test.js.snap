// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="react-multi-carousel-list banners__carousel "
  dir="ltr"
>
  <ul
    className="react-multi-carousel-track "
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "overflow": "unset",
        "transform": "translate3d(0px,0,0)",
        "transition": "none",
      }
    }
  />
</div>
`;
